### PR TITLE
[Core][Testing] adding utilities

### PR DESCRIPTION
**Pull request by philbucher**
I am adding useful utilities, 
I plan to extend them when improving / refactoring the CI-scripts (esp for the MPI-tests)

@maceligueta you could use this in #7093 (if you want, I know this comes too late)

### CODE REVIEW

File: applications/CoSimulationApplication/tests/test_co_sim_io_py_exposure.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 from KratosMultiphysics.CoSimulationApplication import CoSimIO
 from KratosMultiphysics import kratos_utilities as kratos_utils
+from KratosMultiphysics.testing.utilities import GetPython3Command
```
Comment (sunethwarna) [inline applications/CoSimulationApplication/tests/test_co_sim_io_py_exposure.py:6]: why sticking only to `Python3`? May be we can generalize it to `GetPythonCommand` (thinking about future releases)?

Comment (philbucher) [reply applications/CoSimulationApplication/tests/test_co_sim_io_py_exposure.py:6]: I intentionally made it python3, this is what I want, and not (assuming there will be) e.g. Python4, which most probably will behave differently (just like py2 and py3 did)

**Comment by roigcarlo** [comment]
Agree with the changes. 
I am not sure in the uses cases of the script, so I will arise a concern, discard it if I am saying nonsense.

If the co-sim application is used in a system where Kratos that is installed but not compiled we also should take into account that someone my run in a ubunut/etc... with only python 2.x as default and only that version, in that case the getpython command would return an error (an invalid version of python).

To solve that situation in the `run_test.py` script for example we relay on default to `runkratos` instead unless the user override that option manually. Maybe we could take a similar approach for the fallback case here if what I said is a real concern (again, not sure which are going to be the use cases) 

Also this may turn helpful if at some point we start to ship with additional python modules like numpy etc....

**Comment by philbucher** [comment]
thx for the comment @roigcarlo 
I consider this important so lets speak in one of the next meetings to clarify things

**Comment by philbucher** [comment]
@roigcarlo in our call later can we also speak abt this?

**Comment by roigcarlo** [comment]
sure!

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**